apps/ejabberd/src/mongoose_privacy.erl

```diff
@@ -0,0 +1,66 @@
+%%%-------------------------------------------------------------------
+%%% @author bartek <bartlomiej.gorny@erlang-solutions.com>
+%%% @copyright (C) 2017, Erlang Solutions sp. z o.o.
+%%% @doc
+%%%
+%%% @end
+%%% Created : 06. Feb 2017 11:37
+%%%-------------------------------------------------------------------
+-module(mongoose_privacy).
+-author("bartek").
+
+-include("ejabberd.hrl").
+-include("mod_privacy.hrl").
+
+%% API
+-export([privacy_check_packet/6, privacy_check_packet/7]).
+
+%%% API %%%
+
+-spec privacy_check_packet(Acc :: mongoose_acc:t(),
+                           Server :: binary(),
+                           User :: binary(),
+                           PrivacyList :: #userlist{},
+                           To :: ejabberd:jid(),
+                           Dir :: 'in' | 'out') -> {mongoose_acc:t(), allow|deny|block}.
+%% @doc abbreviated version - in most cases we have 'from' in Acc, but sometimes
```

Comment:
Documentation before spec is probably more accurate. Do you know what doc generators expect? Not that we use them often but maybe we can follow their convention.

@erszcz how about your cool [erszcz/docsh](https://github.com/erszcz/docsh)? Do they expect the doc before or after spec? Or it doesn't really matter?